src/Microsoft.AspNet.Mvc.DataAnnotations/DataTypeClientModelValidatorProvider.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    /// <summary>
+    /// An implementation of <see cref="IClientModelValidatorProvider"/> which provides client validators
+    /// for numeric types for which we set type='text'.
+    /// </summary>
+    public class DataTypeClientModelValidatorProvider : IClientModelValidatorProvider
+    {
+        private static readonly HashSet<Type> _numericTypes = new HashSet<Type>(new Type[]
+        {
+            // Including only the types for which we set type='text'.
+            typeof(float), typeof(double), typeof(decimal)
+        });
+
+        /// <inheritdoc />
+        public void GetValidators(ClientValidatorProviderContext context)
+        {
+            var type = context.ModelMetadata.ModelType;
+            var typeToValidate = Nullable.GetUnderlyingType(type) ?? type;
+
+            if (_numericTypes.Contains(typeToValidate))
```

Comment:
Maybe just check the types? It's a small set of types and presumably isn't going to grow